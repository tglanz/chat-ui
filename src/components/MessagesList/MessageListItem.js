import React from 'react';
import { Image, List } from 'semantic-ui-react';
import classnames from 'classnames';

// TBD - Think of adding a timestamp
export default ({ avatar, username, text }) => <List.Item
    className={ classnames(
        "message-list-item", {
        "sent-by-me": username.indexOf("psyduc") !== -1
    }) }
>
    <Image avatar circular src={ avatar } />

    <List.Content>
        <List.Header as='h3'>{ username }</List.Header>
        <List.Description>{ text }</List.Description>
    </List.Content>
</List.Item>

/*

import { Grid, Image, Header } from 'semantic-ui-react';

export default ({ avatar, username, text }) => <div className={ classNames('messages-list-item') }>
    <Grid columns={2}>
            <Grid.Row>
                <Grid.Col width={1}>
                    <Image src={ avatar } />
                </Grid.Col>
                <Grid.Col width={5}>
                    <Header as="h3">{ username }</Header>
                </Grid.Col>
            </Grid.Row>
            <Grid.Row>
                <Grid.Col>
                    <p>{ text }</p>
                </Grid.Col>
            </Grid.Row>
    </Grid>
</div>
*/

