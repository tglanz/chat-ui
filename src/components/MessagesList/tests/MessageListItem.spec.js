import React from 'react';
import renderer from 'react-test-renderer';

import MessageListItem from '../MessageListItem';

describe("MessageListItem", () => {
    it("snapshot", () => {

        const tree = renderer
            .create(<MessageListItem
                username="some username" text="some text" avatar="some avatar"/>)
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
})