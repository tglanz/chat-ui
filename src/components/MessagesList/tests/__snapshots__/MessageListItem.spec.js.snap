// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageListItem snapshot 1`] = `
<div
  className="item message-list-item"
  data-value={undefined}
  onClick={[Function]}
  role="listitem"
>
  <img
    alt={undefined}
    className="ui avatar circular image"
    height={undefined}
    src="some avatar"
    width={undefined}
  />
  <div
    className="content"
  >
    <h3
      className="header"
    >
      some username
    </h3>
    <h6
      className="header"
    >
      a few seconds ago
    </h6>
    <div
      className="description"
    >
      some text
    </div>
  </div>
</div>
`;
